### PR TITLE
Pepperball MK2

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Pepperball, this time instead of TGAS it fires projectiles that drain plasma, thank you splinter.

## Why It's Good For The Game
adds a gun with a unique mechanic nothing else does

## Changelog
:cl:
add: Added new pepperball gun
spriteadd: Pepperballgun and magazine
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -1950,6 +1950,22 @@ datum/ammo/bullet/revolver/tp44
...
+	accurate_range = 15
+	damage_type = STAMINA
+	armor_type = "bio"
+	damage = 70
+	penetration = 0
+
+/datum/ammo/bullet/pepperball/on_hit_mob(mob/living/victim, obj/projectile/proj)
+	if(isxeno(victim))
+		var/mob/living/carbon/xenomorph/X = victim
+		X.use_plasma(50)
```
Comment (BraveMole) [inline code/modules/projectiles/ammo_datums.dm:1967]: magic number

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -1950,6 +1950,22 @@ datum/ammo/bullet/revolver/tp44
...
 
+/datum/ammo/bullet/pepperball
+	name = "pepperball"
+	hud_state = "grenade_frag"
+	hud_state_empty = "battery_empty"
+	flags_ammo_behavior = AMMO_BALLISTIC
+	accurate_range = 15
+	damage_type = STAMINA
+	armor_type = "bio"
+	damage = 70
```
Comment (BraveMole) [inline code/modules/projectiles/ammo_datums.dm:1961]: So it does 70 stam damage on hit on human, that's it?

**Review by BraveMole — CHANGES_REQUESTED** [review]
Sprite stil ugly :'(

**Comment by Kuro-ai** [comment]
The best sprite in the world is back

**Comment by EvilMelonTheSecond** [comment]
Ewwww what's that shade of green(brown?) I love it, please let me attach a scope on it.

File: code/modules/projectiles/magazines/specialist.dm [hunk 3]
```diff
@@ -288,3 +288,16 @@
 	w_class = WEIGHT_CLASS_NORMAL
 	gun_type = /obj/item/weapon/gun/rifle/railgun
 	icon_state_mini = "mag_railgun"
+
+// pepperball
+
+/obj/item/ammo_magazine/pepperball
+	name = "pepperball canister (SAN balls)"
+	desc = "A canister holding a projectile to be used inside a railgun."
```
Comment (Elli-Skala) [inline code/modules/projectiles/magazines/specialist.dm:296]: wouldnt this be used inside a pepperball gun not a railgun?

**Comment by SplinterGP** [comment]
> 
> 
> Sprite stil ugly :'(

lies, the sprite is made by bimmer our supreme spriteman

File: code/modules/projectiles/guns/specialist.dm [hunk 4]
```diff
@@ -1197,3 +1197,34 @@
...
+
+
+// PEPPERBALL GUN
+
+//-------------------------------------------------------
+//TLLL-12
+
+/obj/item/weapon/gun/rifle/pepperball
+	name = "\improper TLLL-12 pepperball gun"
+	desc = "The TLLL-12 is ostensibly riot control device used by the TGMC in spiffy colors, working through a SAN ball that sends a short acting neutralizing chemical to knock out it's target, or weaken them. Guranteed to work on just about everything. Uses SAN Ball Holders as magazines."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/specialist.dm:1209]:
```suggestion
	desc = "The TLLL-12 is ostensibly riot control device used by the TGMC in spiffy colors, working through a SAN ball that sends a short acting neutralizing chemical to knock out its target or weaken them. Guaranteed to work on just about everything, including xenomorph targets. Uses SAN chemical ball holders as magazines."
```
i know you can't be trusted

File: code/modules/projectiles/magazines/specialist.dm (hunk 3 above)
Comment (XSlayer300) [reply code/modules/projectiles/magazines/specialist.dm:296]:
```suggestion
	desc = "A canister meant to hold SAN chemical balls for the pepperball gun.."
```

File: code/modules/projectiles/guns/specialist.dm (hunk 4 above)
Comment (Pariah919) [reply code/modules/projectiles/guns/specialist.dm:1209]: no

**Comment by SilencedMP5A5** [comment]
the gun gives you shrapnel

also doesn't leave behind the purple clouds idk if that's intended or not

also the magazine sprite does not exist and the flavor text is the same as the railgun magazine

**Comment by bowlofcereal** [comment]
i love the water gun that i have to fill with an error magazine

**Comment by Wayland-Smithy** [comment]
This, purple fart gas, and tesla. Really? Where is the xeno spit that melts GPMG ammo still in the mag the hit marine is holding? It'd have the same kind of effect.

**Pull request merged**